src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11TlsKeyMaterialGenerator.java

```diff
@@ -119,6 +119,8 @@ protected void engineInit(AlgorithmParameterSpec params,
             mechanism = CKM_SSL3_KEY_AND_MAC_DERIVE;
         } else if (tlsVersion == 0x0301 || tlsVersion == 0x0302) {
             mechanism = CKM_TLS_KEY_AND_MAC_DERIVE;
+        } else if (tlsVersion == 0x0303) {
+            mechanism = CKM_TLS12_KEY_AND_MAC_DERIVE;
         }
```

Comment:
So this hasn't worked since the TLSv1.2 support was added in 2018?  Ouch!  

I wonder if how many are using PKCS11 for TLS.  Seems like this should have been found earlier.

Double check with @valeriepeng , but looks ok to me...

Reply:
It's not that bad, the crash only happens when you (mis-)use SunTlsKeyMaterial instead of SunTls12KeyMaterial with TLS 1.2. JSSE uses the correct one. I only hit this bug when experimenting with tests.

Reply:
Whew...thanks.